src/components/swap/ConfirmSwapModal.tsx

```diff
@@ -276,7 +276,7 @@ export default function ConfirmSwapModal({
                     ? t`Permit2 allows token approvals to be shared and managed across different applications.`
                     : t`This provides the Uniswap protocol access to your token for trading. For security, this will expire after 30 days.`,
                 button: (
-                  <ButtonPrimary marginX="24px" onClick={startApproveAndSwapFlow}>
+                  <ButtonPrimary marginX="24px" onClick={startApproveAndSwapFlow} data-testid="ConfirmSwapModal-retry">
```

Comment:
keep all your data-testids in lower-cased snake case (eg `like-this`) for consistency